test/sql/update/test_update_issue_3170.test

```diff
@@ -0,0 +1,64 @@
+# name: test/sql/update/test_update_issue_3170.test
+# description: Update indexed table rows with large row_id ( > 122879) which cross the first segment/row_group.
+# group: [update]
+
+statement ok
+PRAGMA enable_verification
+
+# create a table
+statement ok
+CREATE TABLE student(id INTEGER,name VARCHAR,PRIMARY KEY(id));
+
+statement ok
+INSERT INTO student SELECT i, 'creator' FROM RANGE(260001) tbl(i)
+
+# read
+# rowgroup 0
+query I
+SELECT name FROM student WHERE id = 122879
+----
+creator
+
+# rowgroup 1
+query I
+SELECT name FROM student WHERE id = 122881
+----
+creator
+
+# rowgroup 2
+query I
+SELECT name FROM student WHERE id = 245780
+----
+creator
+
+# Update
+# rowgroup 0
+statement ok
+UPDATE student SET name = 'updator0' WHERE id = 122879
+
+# rowgroup 1
+statement ok
+UPDATE student SET name = 'updator1' WHERE id = 122881
+
+# rowgroup 2 
+statement ok
+UPDATE student SET name = 'updator2' WHERE id = 245780
+
```

Comment:
Can we add another test here that reaches into a vector that does not have a `vector_index` of zero? Just for my sanity...
```sql
statement ok
UPDATE student SET name = 'updator3' WHERE id = 150881

query I
SELECT name FROM student WHERE id = 150881
----
updator3
```

Reply:
Thank you Tania, I just added this test.